Pending_updates/cit_031_air_quality_PM25_concentration/cit_031_air_quality_PM25_concentration_processing.py

```diff
@@ -0,0 +1,162 @@
+    sys.path.append(utils_path)
+import util_files
+import util_cloud
+import urllib
+from zipfile import ZipFile
+import ee
+import subprocess
+from google.cloud import storage
+import xarray
+import logging
+
+# Set up logging
+# Get the top-level logger object
+logger = logging.getLogger()
+for handler in logger.handlers: logger.removeHandler(handler)
+logger.setLevel(logging.INFO)
+# make it print to the console.
+console = logging.StreamHandler()
+logger.addHandler(console)
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+# name of asset on GEE where you want to upload data
+# this should be an asset name that is not currently in use
+dataset_name = 'cit_031_air_quality_PM25_concentration' #check
+
+logger.info('Executing script for dataset: ' + dataset_name)
+# create a new sub-directory within your specified dir called 'data'
+# within this directory, create files to store raw and processed data
+data_dir = util_files.prep_dirs(dataset_name)
+    
+'''
+Download data and save to your data directory
+'''
+logger.info('Downloading raw data')
+# insert the url used to download the data from the source website
+url_list = ['http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201101_201112_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201201_201212_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201301_201312_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201401_201412_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201501_201512_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201601_201612_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201701_201712_0p01.nc',
+            'http://fizz.phys.dal.ca/~atmos/datasets/EST2020/ACAG_PM25_GWR_V4GL03_201801_201812_0p01.nc']
+
+# download the data from the source
+raw_data_file = [os.path.join(data_dir,os.path.basename(url)) for url in url_list]
+for url, file in zip(url_list, raw_data_file):
+     urllib.request.urlretrieve(url, file)
+
+'''
+Process data
+'''
+def convert(input_file, output_file):
+    '''
+    Convert netcdf files to tifs 
+    Since the latitude and longitude of this netcdf is reversed, this function first transposes the 'LAT' and 'LON' dimension of the netcdf file before exporting it as a GeoTIFF
+    INPUT   file: file name for netcdf that we want to convert (string)
+            outfile_name: file name for tif we will generate (string)
+    '''
+    # translate the netcdf file into a tif
+    xds = xarray.open_dataset(input_file)
+    PM25 = xds.PM25.transpose('LAT', 'LON')
+    PM25.rio.set_spatial_dims(x_dim="LON", y_dim="LAT", inplace=True)
+    PM25.rio.write_crs("EPSG:4326", inplace=True)
+    PM25.rio.to_raster(output_file[:-4] + '_pre.tif')
```

Comment:
Try to make commenting very detailed so each line can be understood by someone who has never seen a function before - please change any of the comments I added if they aren't correct!
```suggestion
    # open netcdf file
    xds = xarray.open_dataset(input_file)
    # transpose 'LAT' and 'LON' dimensions because they were reversed
    PM25 = xds.PM25.transpose('LAT', 'LON')
    # define the name of the variables to be used for the x and y dimensions in the dataset
    PM25.rio.set_spatial_dims(x_dim="LON", y_dim="LAT", inplace=True)
    # add crs information to data
    PM25.rio.write_crs("EPSG:4326", inplace=True)
    # translate the netcdf file into a tif
    PM25.rio.to_raster(output_file[:-4] + '_pre.tif')
```